pkg/gatherers/clusterconfig/gather_silenced_alerts_test.go

```diff
@@ -0,0 +1,80 @@
+package clusterconfig
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/insights-operator/pkg/utils/marshal"
+)
+
+type mockAlertsClient struct {
+	data []byte
+}
+
+func (c *mockAlertsClient) RestClient(t *testing.T) *rest.RESTClient {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if c.data == nil {
+			w.WriteHeader(http.StatusNotFound)
+		} else {
+			w.WriteHeader(http.StatusOK)
+			_, err := w.Write(c.data)
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+			}
+		}
+	}))
+
+	baseURL, _ := url.Parse(ts.URL)
+	client, err := rest.NewRESTClient(baseURL, "", rest.ClientContentConfig{}, nil, nil)
+	if err != nil {
+		t.Fatalf("failed to create a client: %v", err)
+	}
+
+	return client
+}
+
+func TestGatherSilencedAlerts(t *testing.T) {
+	tests := []struct {
+		name             string
+		mockAlertsClient *mockAlertsClient
+		wantRecords      []record.Record
+		wantErrsCount    int
+	}{
+		{
+			name: "Get silenced alerts successfully",
+			mockAlertsClient: &mockAlertsClient{
+				data: []byte(`[{"status": {"state": "suppressed"}}]`),
+			},
+			wantRecords: []record.Record{
+				{
+					Name: "config/silenced_alerts.json",
+					Item: marshal.RawByte(`[{"status": {"state": "suppressed"}}]`),
+				},
+			},
+			wantErrsCount: 0,
```

Comment:
Suggestion. It would be more robust to check the particular error messages than just number of errors.

Reply:
I added the test, but, TBH, in this specific case, I don't see much value/difference between checking the error type or the number of errors. 🤔